### PR TITLE
INTMDB-448: custom_db_role error 

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

custom_db_role error where updates skipped during update
Customer had an issue related [resource: mongodbatlas_custom_db_role](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/custom_db_role) in [PRODTRIAGE-3263](https://jira.mongodb.org/browse/PRODTRIAGE-3263).

Link to any related issue(s):

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Review by Zuhairahmed — APPROVED** [review]
LGTM. small, but mighty bug fix!

File: mongodbatlas/data_source_mongodbatlas_custom_db_roles_test.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func TestAccConfigDSCustomDBRoles_basic(t *testing.T) {
 
 					// Test for Data source
 					resource.TestCheckResourceAttrSet(dataSourceName, "project_id"),
-					resource.TestCheckResourceAttr(dataSourceName, "results.#", "1"),
```
Comment (andreaangiolillo) [inline mongodbatlas/data_source_mongodbatlas_custom_db_roles_test.go:40]: [q] Why did we pass "1" here?

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go [hunk 2]
```diff
@@ -101,7 +102,13 @@ func resourceMongoDBAtlasCustomDBRole() *schema.Resource {
 	}
 }
 
+var (
+	customRoleLock sync.Mutex
+)
+
 func resourceMongoDBAtlasCustomDBRoleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	customRoleLock.Lock()
+	defer customRoleLock.Unlock()
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_custom_db_role.go:111]: [q] Could you give me more details about why we hit this issue? Thanks

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: mongodbatlas/data_source_mongodbatlas_custom_db_roles_test.go (hunk 1 above)
Comment (martinstibbe) [reply mongodbatlas/data_source_mongodbatlas_custom_db_roles_test.go:40]: When I fixed concurrency issue other acceptance tests in parallel would increase count > 1 and fail test

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go (hunk 2 above)
Comment (martinstibbe) [reply mongodbatlas/resource_mongodbatlas_custom_db_role.go:111]: Atlas API if you run each terraform resource apply sequentially in slow motion i.e. debugger it updates state correctly
If you run all in parallel API does not come back with consistent values Zuhair's example test 3 separate roles that fail if run all together

Comment (Zuhairahmed) [reply mongodbatlas/resource_mongodbatlas_custom_db_role.go:111]: @andreaangiolillo brief example here: https://drive.google.com/file/d/1aDfPwxjXm35qkEQgP135M6FxZAss5Qlv/view
also in body of jira ticket as well if you need: https://jira.mongodb.org/browse/INTMDB-448

**Pull request merged**